### ISSUE TITLE
Support Variadic arguments

### ISSUE THREAD

**Issue by Taluu** [phpactor/phpactor#621]
With the following piece of code :

```php
<?php

function (Foo ...$foos): void 
{
    foreach ($foos as $foo) {
    }
}
```

`$foo` is not recognized as a `Foo`, but `unknown`. And `$foos` is not recognized as an array of `Foo`s, but a `Foo`. We have to add the `/** @var Foo $foo */` in top of the foreach for that.

Having that would be great. :}

**Comment by dantleech** [comment]
Would be quite simple:

1. The parameter is resolved here: https://github.com/phpactor/worse-reflection/blob/master/lib/Core/Inference/FrameBuilder/FunctionLikeWalker.php#L85
2. We just need to add support to the [SymbolContextResolver](https://github.com/phpactor/worse-reflection/blob/master/lib/Core/Inference/SymbolContextResolver.php) to return a symbol with type `array<SomeType>`

**Comment by dantleech** [comment]
Fixed in #681
